File: LeetCode/Easy/0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (isLeaf(root)) return 0;
        return find(0, root);
    }

    private int find(int sum, TreeNode node) {
        
        // left
        if (node.left != null) {
            if(isLeaf(node.left)) {
                // System.out.println(node.left.val);
                sum += node.left.val;
            } else {
                sum = find(sum, node.left);
            }
        }

        // right
        if (node.right != null) {
            sum = find(sum, node.right);
        }

        return sum;
    }

    private boolean isLeaf(TreeNode node) {
        return node.left == null && node.right == null;
    }
}

File: LeetCode/Easy/0404-sum-of-left-leaves/0404-sum-of-left-leaves.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun sumOfLeftLeaves(root: TreeNode?): Int {

        if (isLeaf(root)) return 0;
        return find(0, root);
    }

    private fun find(sum: Int, node: TreeNode?): Int {
        var add = sum;
        // left
        if (node?.left != null) {
            if(isLeaf(node.left)) {
                add += node.left.`val` ?: 0;
            } else {
                add = find(add, node.left);
            }
        }

        // right
        if (node?.right != null) {
            add = find(add, node.right);
        }

        return add;
    }

    private fun isLeaf(node: TreeNode?): Boolean {
        return node?.left == null && node?.right == null;
    }
}